src/try_fold.rs

```diff
@@ -160,30 +170,25 @@ impl<'a, I, O: Clone, E> FromIterator<TryFold<'a, I, O, E>> for TryFold<'a, I, O
 
 #[cfg(test)]
 mod tests {
+  use std::cell::RefCell;
+
   use super::TryFold;
   use crate::valid::{Valid, ValidationError};
 
   #[test]
   fn test_and() {
+    let call_number = RefCell::new(0);
     let t1 = TryFold::<i32, i32, ()>::new(|a: &i32, b: i32| Valid::succeed(a + b));
-    let t2 = TryFold::<i32, i32, ()>::new(|a: &i32, b: i32| Valid::succeed(a * b));
-    let t = t1.and(t2);
-
-    let actual = t.try_fold(&2, 3).to_result().unwrap();
-    let expected = 10;
-
-    assert_eq!(actual, expected)
-  }
-
-  #[test]
-  fn test_combine_ok() {
-    let t1 = TryFold::<i32, i32, ()>::new(|a: &i32, b: i32| Valid::succeed(a + b));
-    let t2 = TryFold::<i32, i32, ()>::new(|a: &i32, b: i32| Valid::succeed(a * b));
+    let t2 = TryFold::<i32, i32, ()>::new(|a: &i32, b: i32| {
+      *call_number.borrow_mut() += 1;
+      Valid::succeed(a * b)
+    });
     let t = t1.and(t2);
 
     let actual = t.try_fold(&2, 3).to_result().unwrap();
     let expected = 10;
 
+    assert_eq!(*call_number.borrow(), 1);
```

Comment:
t2 is called 2 times rn

Reply:
Can we add a separate test for asserting FnOnce behaviour? The existing tests should pass with the new implementation.

Reply:
actually, removed that change, since `test_order` should already verify that the handlers are called once